include/seastar/core/memory.hh

```diff
@@ -382,6 +383,25 @@ public:
     void operator=(scoped_large_allocation_warning_disable&&) = delete;
 };
 
+struct allocation_site {
+    mutable size_t count = 0; // number of live objects allocated at backtrace.
+    mutable size_t size = 0; // amount of bytes in live objects allocated at backtrace.
+    mutable const allocation_site* next = nullptr;
+    simple_backtrace backtrace;
+
+    bool operator==(const allocation_site& o) const {
+        return backtrace == o.backtrace;
+    }
+
+    bool operator!=(const allocation_site& o) const {
```

Comment:
In C++20 we don't need the != anymore due to spaceship, but seastar supports the last _two_ versions which until perhaps recently was 17 + 20, so this is probably needed to support C++17. However, I believe they may have flipped recently to 20 + 23. Nothing to do here, just more informational.